child_compassion/model/project_compassion.py

```diff
@@ -184,6 +216,37 @@ def _update_country(self, cr, uid, country_code, context=None):
             country_obj.update_informations(cr, uid, [country_id], context)
         return country_id
 
+    def _get_age_groups(self, cr, uid, project, context=None):
+        """ Get age group from compassion webservice and l
+            Returns id of generated age_group or None if failed """
+        # Delete old age_groups
+        for age_group in project.age_group_ids:
+            project.write({'age_group_ids': [(2, age_group.id)]})
+        json_data = self._cornerstone_fetch(project.code + '/agegroups',
+                                            'cdspimplementors')
+        value_obj = self.pool.get('compassion.translated.value')
+        res = list()
+        for group in json_data['projectAgeGroupCollection']:
+            values = list()
+            vals = {
+                'project_id': project.id,
+                'low_age': group['lowAge'],
+                'high_age': group['highAge'],
+                'school_hours': group['schoolHours'],
+            }
+            values.extend(value_obj.get_value_ids(
+                cr, uid, group['schoolDays'].split(','),
+                'school_days', context))
+            values.extend(value_obj.get_value_ids(
+                cr, uid, group['schoolMonths'].split(','),
+                'school_months', context))
+
+            vals['school_days_ids'] = [(6, 0, values)]
+            age_project_obj = self.pool.get('compassion.project.age.group')
```

Comment:
You can declare this variable before line 229, so that it is not initialized in each iteration of the for statement.